panda/src/cocoadisplay/cocoaGraphicsWindow.mm

```diff
@@ -568,20 +568,45 @@
   }
 
   if (_properties.has_cursor_filename()) {
-    NSImage *image = load_image(_properties.get_cursor_filename());
+    NSData *image_data = load_image_data(_properties.get_cursor_filename());
+
+    // Read the metadata from the image, which should contain hotspotX and
+    // hotspotY properties.
+    CGImageSourceRef cg_image = CGImageSourceCreateWithData((CFDataRef)image_data, nullptr);
```

Comment:
Oh, and `load_image_data` may return null, but if it does, this call will probably crash.